src/resolve/resolved-dnstls-gnutls.c

```diff
@@ -236,7 +236,9 @@ int dnstls_manager_init(Manager *manager) {
 
         r = gnutls_certificate_allocate_credentials(&manager->dnstls_data.cert_cred);
         if (r < 0)
-                return -ENOMEM;
+                return log_warning_errno(SYNTHETIC_ERRNO(EIO),
```

Comment:
EIO is a bit weird for that, no?

I like ENOTRECOVERABLE for things that are otherwise hard to categorize